### PR TITLE
ui: create object list input component

### PR DESCRIPTION
**Pull request by hellobontempo**
New component that allows multiple inputs in a single row, each row is its own object. The on change action returns an array of objects to the parent.
object-list-input

<hr>

### The component accommodates a variable number of inputs:

Screen Shot 2023-01-11 at 11 18 37 AM

Screen Shot 2023-01-11 at 11 19 39 AM

### CODE REVIEW

File: ui/lib/core/addon/components/object-list-input.hbs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          {{on "click" this.addRow}}
+        >
+          Add
+        </button>
+      {{else}}
+        <button
+          data-test-object-list-delete-button={{index}}
+          type="button"
+          class="button is-icon"
+          {{on "click" (fn this.removeRow index)}}
```
Comment (hashishaw) [inline ui/lib/core/addon/components/object-list-input.hbs:40]: can we add an aria-label for non-sighted friends? 
```suggestion
          aria-label="remove"
          {{on "click" (fn this.removeRow index)}}
```

**Review by hashishaw — APPROVED** [review]
Great work! Just one small change request

File: ui/lib/core/addon/components/object-list-input.hbs (hunk 1 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/object-list-input.hbs:40]: ah yes! thank you!

File: ui/lib/core/addon/components/object-list-input.hbs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      {{/let}}
+    {{/each-in}}
+    <div class="control is-align-end">
+      {{#if (eq (inc index) this.inputList.length)}}
+        <button
+          data-test-object-list-add-button
+          type="button"
+          class="button is-outlined is-primary"
+          disabled={{this.disableAdd}}
+          {{on "click" this.addRow}}
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/object-list-input.hbs:31]:
```suggestion
          aria-label="add"
          {{on "click" this.addRow}}
```

File: ui/lib/core/addon/components/object-list-input.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  assertKeys(arrayOfObjects, requiredKeys) {
+    const argName = requiredKeys.includes('label') ? '@objectKeys' : '@inputValue';
+    return assert(
+      `objects in the ${argName} array must include keys called: ${requiredKeys.join(', ')}`,
+      arrayOfObjects.every((obj) => requiredKeys.every((k) => Object.keys(obj).includes(k)))
+    );
+  }
+
+  createEmptyRow(keys) {
+    // create a new object with from input keys with empty values
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/object-list-input.js:55]:
```suggestion
    // create a new object from input keys that have empty values
```

**Pull request merged**